### PR TITLE
get_file(): skip if file with same checksum exists

### LINKED ISSUE #109: Get file from backend only if checksums differ
In `Backend.put_file()` we already check if the checksum of a local and backend file is the same and in that case we skip the operation. We can do the same with `Backend.get_file()`.

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes #109 

image

### CODE REVIEW

File: audbackend/core/backend.py [hunk 1]
```diff
@@ -263,19 +266,23 @@ def get_file(
 
         audeer.mkdir(dst_root)
         if (
-                not os.access(dst_root, os.W_OK) or
-                (os.path.exists(dst_path) and not os.access(dst_root, os.W_OK))
+            not os.access(dst_root, os.W_OK) or
+            (os.path.exists(dst_path) and not os.access(dst_root, os.W_OK))
```
Comment (hagenw) [inline audbackend/core/backend.py:270]: Why do we need the second line? If this is true, the first line should always be true as well.

**Review by hagenw — COMMENTED** [review]
What did we before, skip if file exists or overwrite it?

**Comment by frankenjoe** [comment]
> What did we before, skip if file exists or overwrite it?

We always overwrite the file even if the files were identical. Now we only overwrite if the checksums do not match.

File: audbackend/core/backend.py (hunk 1 above)
Comment (frankenjoe) [reply audbackend/core/backend.py]: Thanks for spotting, the second line should check if we have permission to `dst_path`, so I fixed it to:

```
(os.path.exists(dst_path) and not os.access(dst_path, os.W_OK))
```

I also added a test for it now.

**Pull request merged**